server/broadcast_test.go

```diff
@@ -1818,26 +1993,21 @@ func genBcastSess(t *testing.T, url string, os drivers.OSSession, mid core.Manif
 		},
 	})
 	require.Nil(t, err, fmt.Sprintf("Could not marshal results for %s", url))
-	ts, mux := stubTLSServer()
-	mux.HandleFunc("/segment", func(w http.ResponseWriter, r *http.Request) {
+	transcoderURL := stubTestTranscoder(ctx, func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
 		w.Write(buf)
 	})
-	defer func() {
-		// Work around a weird timing issue. Tests fail if the server closes
-		// in-scope (prob leads to something like the client being unable
-		// to read the response?), so we delay the close for a little bit
-		go func() {
-			// We assume this test doesn't take more than 1s
-			// But if it does (eg, we get a POST error), then bump this up
-			time.Sleep(1 * time.Second)
-			ts.Close()
-		}()
-	}()
 	return &BroadcastSession{
 		Broadcaster:      stubBroadcaster2(),
 		Params:           &core.StreamParameters{ManifestID: mid, Profiles: []ffmpeg.VideoProfile{ffmpeg.P144p30fps16x9}, OS: os},
 		BroadcasterOS:    os,
-		OrchestratorInfo: &net.OrchestratorInfo{Transcoder: ts.URL, AuthToken: stubAuthToken},
+		OrchestratorInfo: &net.OrchestratorInfo{Transcoder: transcoderURL, AuthToken: stubAuthToken},
 	}
 }
+
+func stubTestTranscoder(ctx context.Context, handler http.HandlerFunc) string {
+	ts, mux := stubTLSServer()
+	go func() { <-ctx.Done(); ts.Close() }()
```

Comment:
Does waiting for ctx.Done() here fix the timing issue mentioned in the now deleted defer code block that we previously worked around by adding a hardcoded sleep?

Reply:
Yeah! Since this function will be called with a `context.Context` that is only (and immediately) cancelled when the respective test is done, this will hold the server alive as long as the test is running i.e. while the server might be in use. 

It might take a little longer to close in case of multiple test iterations on the same function that creates separate test servers, which is the case for the metadata queue test, but I don't think we need to optimize for those (we could create and cancel separate contexts for each iteration). Closing them all after the test function is finished seems good enough to me.